cpp/src/arrow/ipc/reader.cc

```diff
@@ -243,6 +247,7 @@ class ArrayLoader {
   }
 
   Status GetFieldMetadata(int field_index, ArrayData* out) {
+    DCHECK_NE(nullptr, out_);
```

Comment:
We don't use `out_` in this method.

Reply:
oops, I've change the code, and push check in:
1. Before `Visit` the subtypes
2. `LoadXXX` called.